woocommerce/woocommerce-gateway-stripe
# Only load Stripe JS on product page when required

## Changes proposed in this Pull Request:

Fixes #2065 

- Don't load Stripe JS on product pages when PRBs are disabled.
- Don't load Stripe JS on the cart page when PRBs are disabled.

## Testing instructions

> **Note:** The test should be performed with UPE enabled and disabled, and on both the Shortcode and Block-based cart pages.

1. Make sure PRBs are only present on the **Checkout** page.
2. `git switch develop`.
3. Open the **Network** tab of your browser's developer tools.
4. Open any product details page.
5. Notice that we load scripts from js.stripe.com.
6. Add a product to your cart and go to your store's cart page.
7. Notice that we load scripts from js.stripe.com.
8. `git switch fix/2065-only-load-stripe-js-on-product-pages-when-required`
9. Open any product details page.
10. Make sure we still load scripts from js.stripe.com.
11. Add a product to your cart and go to your store's cart page.
12. Make sure we still load scripts from js.stripe.com.
13. Use the Code Snippets plugin to add the following filters:

```php
add_filter( 'wc_stripe_load_scripts_on_product_page_when_prbs_disabled', '__return_false' );
add_filter( 'wc_stripe_load_scripts_on_cart_page_when_prbs_disabled', '__return_false' );
```

14. Open any product details page.
15. Make sure no scripts are loaded from js.stripe.com
16. Add a product to your cart and go to your store's cart page.
17. Make sure no scripts are loaded from js.stripe.com


---


**Post merge**
